Application/EdFi.Ods.SandboxAdmin/wwwroot/js/vendor.js

```diff
@@ -80,40 +90,75 @@ var confirmationDialog = function () {
 
 var addApplicationDialog = function () {
     var self = this;
-    self.vendorName = ko.observable().extend({
-        required: true,
+
+    self.fieldsTouched = ko.observable(false);
+
+    self.buttonText = ko.observable("");
+
+    self.vendorName = ko.observable(null).extend({
+        required: {
+            message: 'The Vendor Name field is required.',
+            onlyIf: function () { return self.fieldsTouched(); }
+        },
         pattern: {
             message: 'Please enter only letters',
-            params: /^[A-Za-z]+$/
+            params: /^[A-Za-z]+$/,
+            onlyIf: function () { return self.fieldsTouched(); }
```

Comment:
This line should be removed so that users can immediately see a validation error upon entering a vendor name containing non-letter character(s). This change will make the validation behavior consistent with that in the "Add Clent" and "Add Application" dialogs.

Reply:
As Discussed , Validation is needed when user clicks the Add button , So all validation happens only when you click Add button for both screens